src/FundSourceApprover/FundSourceAllowerAdmin.sol

```diff
@@ -0,0 +1,115 @@
+        _transferOwnership(_owner);
+        nativeFundDestination = _nativeFundDestination;
+        erc20FundDestination = _erc20FundDestination;
+    }
+
+    /// @notice create fund source allower that handles the native currency
+    /// @param _sourceChainId chain ID of fund source, or other unique identifier (used for bookkeeping)
+    function createNativeFundSourceAllower(
+        uint256 _sourceChainId
+    ) external onlyOwner {
+        NativeFundSourceAllower allower = new NativeFundSourceAllower{
+            salt: _getSalt(_sourceChainId, address(0))
+        }(_sourceChainId, nativeFundDestination, address(this));
+        emit NewFundSourceAlowerCreated({
+            addr: address(allower),
+            chainId: _sourceChainId,
+            token: address(0)
+        });
+    }
+
+    /// @notice create fund source allower that handles an ERC20
+    /// @param _sourceChainId chain ID of fund source, or other unique identifier (used for bookkeeping)
+    /// @param _token address on this chain of token that fund source allower will handle
+    function createErc20FundSourceAllower(
+        uint256 _sourceChainId,
+        address _token
+    ) external onlyOwner {
+        if (!Address.isContract(_token)) {
+            revert NotAContract(_token);
+        }
+        Erc20FundSourceAllower allower = new Erc20FundSourceAllower{
+            salt: _getSalt(_sourceChainId, _token)
+        }(_sourceChainId, erc20FundDestination, address(this), _token);
+        emit NewFundSourceAlowerCreated({
+            addr: address(allower),
+            chainId: _sourceChainId,
+            token: _token
+        });
+    }
+
+    /// @notice Toggle approved for target allower; enables/ disables transfer of funds to destination
+    function toggleApprove(address payable _allower) external onlyOwner {
+        bool allowed = FundSourceAllowerBase(_allower).toggleApproved();
+        emit ApprovedToggled(allowed, _allower);
+    }
+
+    /// @notice determine address of a native fund souce allower given an id
+    /// @param _sourceChainId ID of fund source, or other unique identifier
+    function getNativeFundSourceAllowerCreate2Address(
+        uint256 _sourceChainId
+    ) public view returns (address) {
+        bytes32 salt = _getSalt(_sourceChainId, address(0));
+        bytes32 bytecodeHash = keccak256(
+            abi.encodePacked(
+                type(NativeFundSourceAllower).creationCode,
+                abi.encode(_sourceChainId, nativeFundDestination, address(this))
+            )
+        );
+        return Create2.computeAddress(salt, bytecodeHash);
+    }
+
+    /// @notice determine address of am erc20 fund souce allower given its id and token
+    /// @param _sourceChainId ID of fund source, or other unique identifier
+    /// @param _token token of erc20 fund souce allower
+    function getErc20FundSourceAllowerCreate2Address(
+        uint256 _sourceChainId,
+        address _token
```

Comment:
also add the isContract check here?

Reply:
this is meant for computing the address even if it isn't yet deployed